Core/include/Acts/TrackFitting/detail/GsfSmoothing.hpp

```diff
@@ -60,14 +61,18 @@ auto bayesianSmoothing(component_iterator_t fwdBegin,
 
       const auto new_weight = std::exp(-0.5 * exponent) * weight_a * weight_b;
 
-      if (new_weight == 0) {
-        return ResType(GsfError::SmoothingFailed);
+      if (not std::isfinite(new_weight) or new_weight < weightCutoff ) {
+        continue;
       }
 
       smoothedState.push_back({new_weight, new_pars, new_cov});
```

Comment:
Since we don't return the error anymore, do you think it could make the code more readable to pull the `push_back` into an inverted `if`?

```suggestion
      if (std::isfinite(new_weight) and new_weight > weightCutoff ) {
        smoothedState.push_back({new_weight, new_pars, new_cov});
      }
```